cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -493,9 +493,17 @@ public void programHistory(HttpRequest request, HttpResponder responder,
                              @QueryParam("end") String endTs,
                              @QueryParam("limit") @DefaultValue("100") final int resultLimit)
     throws Exception {
-    programHistoryVersioned(request, responder, namespaceId, appName,
-                   getLatestAppVersion(new NamespaceId(namespaceId), appName), type,
-                   programName, status, startTs, endTs, resultLimit);
+    ProgramType programType = getProgramType(type);
+
+    long start = (startTs == null || startTs.isEmpty()) ? 0 : Long.parseLong(startTs);
+    long end = (endTs == null || endTs.isEmpty()) ? Long.MAX_VALUE : Long.parseLong(endTs);
```

Comment:
In the handler, we need to assume the user can pass in anything they want into the arguments, and should be validating accordingly. If any of the long parsing or ProgramRunStatus parsing throws an exception, we should return a 400 with descriptive message. For example, something like:

```
long start = parseAndValidate(startTs, 0, "start");
long end = parseAndValidate(endTs, Long.MAX_VALUE : "end");
if (end < start) {
  throw new BadRequestException(String.format("Invalid end time %d. It must be greater than the start time %d", end, start)); 
}
ProgramRunStatus ....

private long parseAndValidate(String strVal, long defaultVal, String paramName) {
  if (strVal == null || strVal.isEmpty()) {
    return defaultVal;
  }
  try {
    return Long.parseLong(strVal);
  } catch (NumberFormatException e) {
    throw new BadRequestException(String.format("Invalid value %s for %s", strVal, paramName), e);
  }
} 
```